### PR TITLE
New script to generate the command docs

### PR DESCRIPTION
**Pull request by Shellyber**
## Related Issues
fixes: https://jira-dc.paloaltonetworks.com/browse/CIAC-12077
https://jira-dc.paloaltonetworks.com/browse/CIAC-12078

## Description
- Script to generate the command description and the options if the command setup has been changed.
- Pre-commit hook to run the script.

### CODE REVIEW

File: demisto_sdk/scripts/generate_commands_docs/pre_commit_generate_docs.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+
+def extract_changed_commands(modified_files):
+    """Extract command names from modified _setup.py files."""
+    changed_commands = []
+    for file in modified_files:
+        if file.endswith("_setup.py"):
+            command_name = Path(file).stem.replace("_setup", "")
+            changed_commands.append(command_name)
+    return changed_commands
```
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply demisto_sdk/scripts/generate_commands_docs/pre_commit_generate_docs.py:29]: True for all functions

File: demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+import sys
+from pathlib import Path
+from typing import List
+
+from typer.main import get_command
+
+from demisto_sdk.__main__ import app
+
+
+def extract_changed_commands(modified_files: List[str]) -> List[str]:
```
Comment (ShahafBenYakir) [inline demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py:11]: Isn't this duplicate?

File: demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    for param in command.params:
+        param_name = (
+            f"--{param.name.replace('_', '-')}"
+            if param.param_type_name == "option"
+            else param.name
+        )
+        options_text += (
+            f"- **{param_name}**: {param.help or 'No description provided'}\n"
+        )
+        if param.default is not None:
```
Comment (ShahafBenYakir) [inline demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py:56]: why `is not None:`? Isn't `if param.default` enough? if for example it is`''` you still don't want the condition to apply right?

File: demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    command = click_app.commands.get(command_name)  # type: ignore[attr-defined]
+
+    if command is None:
+        return f"No README found for command: {command_name}"
+    return command
+
+
+def get_command_description(command_name: str) -> str:
+    """Retrieve the description (docstring) for the command."""
+    command = get_sdk_command(command_name)
```
Comment (ShahafBenYakir) [inline demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py:33]: What if the response is `"No README found for command..."`?

File: demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+    # Write the updated content back into the README.md
+    with command_doc_path.open("w") as f:
+        f.write(updated_readme)
+
+    print(f"Description and options section updated for command: {command_name}")  # noqa: T201
+
+
+def generate_docs_for_command(command_name: str):
+    """Generate documentation for a specific command."""
+    description = get_command_description(command_name)
```
Comment (ShahafBenYakir) [inline demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py:111]: Are we ok with the description being `'No description provided'`? Where is it going be added to?

File: demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+        options_text += "\n"
+    return options_text
+
+
+def update_readme(command_name: str, description: str, options: str):
+    """Update the README.md file for the command with the given description and options."""
+    command_doc_path = Path("demisto_sdk/commands") / command_name / "README.md"
+
+    if not command_doc_path.exists():
+        print(f"README.md not found for command: {command_name}")  # noqa: T201
```
Comment (ShahafBenYakir) [inline demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py:67]: Why not generate a new one?

File: demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+        f.write(updated_readme)
+
+    print(f"Description and options section updated for command: {command_name}")  # noqa: T201
+
+
+def generate_docs_for_command(command_name: str):
+    """Generate documentation for a specific command."""
+    description = get_command_description(command_name)
+    options = get_command_options(command_name)
+    update_readme(command_name, description, options)
```
Comment (ShahafBenYakir) [inline demisto_sdk/scripts/generate_commands_docs/generate_commands_docs.py:113]: Is it enough to write the file? what about committing etc?

**Review by ShahafBenYakir — COMMENTED** [review]
Nice.

1. Why did we create two Python files?
2. I want us to run multiple tests on this i.e make all sorts of changes - delete, add modified etc as creative as you can bee and fix issues.
3. Do we have infra for unit tests?

**Comment by Shellyber** [comment]
@ShahafBenYakir Doing some more tests and will add tests for the script. But please review again in the meantime

File: demisto_sdk/scripts/generate_commands_docs/pre_commit_generate_docs.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply demisto_sdk/scripts/generate_commands_docs/pre_commit_generate_docs.py]: @Shellyber Please add to your docsting the arguments documentation

**Review by ShahafBenYakir — APPROVED** [review]

File: .changelog/4664.yml [hunk 8]
```diff
@@ -0,0 +1,4 @@
+changes:
+- description: Added new script and a pre-commit hook to generate commands documentation.
```
[the remark this answers is not in the document]
Comment (Shellyber) [reply .changelog/4664.yml:2]: This is why I put it as an internal. 
If you think it is completely not necessary I can remove.

[the remark this answers is not in the document]
Comment (ShahafBenYakir) [reply .changelog/4664.yml:2]: Oh I didn't see that. cool

**Pull request merged**